firefox-ios/Client/Frontend/Home/Wallpapers/v1/Interface/WallpaperManager.swift

```diff
@@ -77,33 +94,31 @@ class WallpaperManager: WallpaperManagerInterface, FeatureFlaggable {
         return true
     }
 
-    /// Determines whether the wallpaper settings can be shown
-    var canSettingsBeShown: Bool {
-        guard hasEnoughThumbnailsToShow else { return false }
-
-        return true
-    }
-
-    /// Returns true if the metadata & thumbnails are available
-    private var hasEnoughThumbnailsToShow: Bool {
-        let thumbnailUtility = WallpaperThumbnailUtility(with: networkingModule)
-
-        guard thumbnailUtility.areThumbnailsAvailable else { return false }
-
-        return true
+    /// Update the value referenced for wallpaper onboarding when
+    /// wallpaper onboarding was shown
+    func onboardingSeen() {
+        userDefaults.set(true, forKey: PrefsKeys.Wallpapers.OnboardingSeenKey)
     }
 
     /// Sets and saves a selected wallpaper as currently selected wallpaper.
     ///
     /// - Parameter wallpaper: A `Wallpaper` the user has selected.
-    public func setCurrentWallpaper(
+    func setCurrentWallpaper(
         to wallpaper: Wallpaper,
         completion: @escaping (Result<Void, Error>) -> Void
     ) {
         do {
             let storageUtility = WallpaperStorageUtility()
             try storageUtility.store(wallpaper)
-
+            if featureFlags.isFeatureEnabled(.homepageRebuild, checking: .buildOnly) {
+                let wallpaperConfig = WallpaperConfiguration(wallpaper: wallpaper)
+                let action = WallpaperAction(
+                    wallpaperConfiguration: wallpaperConfig,
+                    windowUUID: .unavailable,
```

Comment:
Is this `.unavailable` because we don't want to tie the manager to a specific window? I'm just trying to understand if there's a downside to tying the manager to a specific window, or if there's an optimization here where, on iPad, for example, not all windows will have to necessarily attempt to process that action (which is what happens if `.unavailable` is passed in to a reducer)

Reply:
honestly I am not sure I totally understand what it means for an action to send a window uuid. My naive logic here was the wallpaper manager is independent of the view layer and thus should not care about what the current window is. If our separation of concerns is correct the wallpaper manager should only be interfacing with the middleware and the middleware is responsible for firing actions that the view state cares about?

Reply:
Yep, I think that makes perfect sense to me.